scrapy/pipelines/media.py

```diff
@@ -1,12 +1,15 @@
 import functools
+from inspect import signature
 import logging
 from collections import defaultdict
 from twisted.internet.defer import Deferred, DeferredList
 from twisted.python.failure import Failure
+from warnings import warn
```

Comment:
I think this import should be before the Twisted ones? @Gallaecio you are the expert on this :)

Reply:
Hmm, I think it does! Maybe I could also add an empty line between standard library imports and Twisted imports as specified in the [PEP8 guidelines](https://www.python.org/dev/peps/pep-0008/#:~:text=You%20should%20put%20a%20blank%20line%20between%20each%20group%20of%20imports.&text=Standard%20library%20code%20should%20avoid,been%20removed%20in%20Python%203.)?. What say @Gallaecio?

Reply:
And `inspect` should be among other standard library `from` imports (`right after `from collections …`), we group those right after straight imports (without an empty line, though).